asserts/database.go

```diff
@@ -176,3 +192,94 @@ func (db *Database) Check(assert Assertion) error {
 	}
 	return fmt.Errorf("failed signature verification: %v", lastErr)
 }
+
+func (db *Database) readAssertion(assertType AssertionType, primaryPath string) (Assertion, error) {
+	encoded, err := db.readEntry(assertionsRoot, string(assertType), primaryPath)
+	if os.IsNotExist(err) {
+		return nil, ErrNotFound
+	}
+	if err != nil {
+		return nil, fmt.Errorf("broken assertion storage, failed to read assertion: %v", err)
+	}
+	assert, err := Decode(encoded)
+	if err != nil {
+		return nil, fmt.Errorf("broken assertion storage, failed to decode assertion: %v", err)
+	}
+	return assert, nil
+}
+
+func (db *Database) readCurrentRevision(assertType AssertionType, primaryPath string) (int, error) {
+	assert, err := db.readAssertion(assertType, primaryPath)
+	if err == ErrNotFound {
+		return -1, nil
```

Comment:
@niemeyer if we return 0 adding the first revision (which is also 0) would fail, I'm marking not found with (-1, nil) but if you prefer to propagate an explicit NotFound and have bit more logic in the caller works for me